### DISCUSSION TITLE
Why is "fortran" censored in the log display?

### DISCUSSION THREAD

**Discussion by mathomp4** [community/community#27210] [Actions]
<p>This is an odd one, but I noticed it today in trying to get GitHub Actions working on a repo. Namely, when I watch the log output scroll by I see:</p>
<pre><code class="lang-auto">  make verify ESMF_COMM=openmpi ESMF_COMPILER=g***
  shell: sh -e {0}
  env:
    OMPI_ALLOW_RUN_AS_ROOT: 1
    OMPI_ALLOW_RUN_AS_ROOT_CONFIRM: 1
    OMPI_MCA_btl_vader_single_copy_mechanism: none
    FC: g***
    CC: gcc
    CXX: g++
</code></pre>
<p>for this bit of the yaml file:</p>
<pre><code class="lang-yaml">    env:
      OMPI_ALLOW_RUN_AS_ROOT: 1
      OMPI_ALLOW_RUN_AS_ROOT_CONFIRM: 1
      OMPI_MCA_btl_vader_single_copy_mechanism: none
      FC: gfortran
      CC: gcc
      CXX: g++
</code></pre>
<p>and in one of the steps I have:</p>
<pre><code class="lang-yaml">      - name: Versions etc.
        run: |
          gcc --version
          gfortran --version
          mpirun --version
          mpicc --show
          mpic++ --show
          mpifort --show
          echo $BASEDIR
          git submodule status
</code></pre>
<p>and the output is:</p>
<pre><code class="lang-auto">gcc (GCC) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

GNU Fortran (GCC) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

mpirun (Open MPI) 4.0.5

Report bugs to http://www.open-mpi.org/community/help/
gcc -I/opt/openmpi-4.0.5/include -pthread -Wl,-rpath -Wl,/opt/openmpi-4.0.5/lib -Wl,--enable-new-dtags -L/opt/openmpi-4.0.5/lib -lmpi
g++ -I/opt/openmpi-4.0.5/include -pthread -Wl,-rpath -Wl,/opt/openmpi-4.0.5/lib -Wl,--enable-new-dtags -L/opt/openmpi-4.0.5/lib -lmpi
g*** -I/opt/openmpi-4.0.5/include -pthread -I/opt/openmpi-4.0.5/lib -Wl,-rpath -Wl,/opt/openmpi-4.0.5/lib -Wl,--enable-new-dtags -L/opt/openmpi-4.0.5/lib -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lmpi
</code></pre>
<p>And then later on in the logging:</p>
<pre><code class="lang-auto">/bin/bash ../../libtool --tag=CXX  --tag=CC   --mode=link mpicc  -fPIC   -fopenmp -std=c99 -D_BSD_SOURCE -D_POSIX_SOURCE  -L/__w/ESMA-Baselibs/x86_64-pc-linux-gnu/g***/Linux/lib -lnetcdf -ljpeg -lmfhdf -ldf -ljpeg -lhdf5_hl -lhdf5 -lm -L/__w/ESMA-Baselibs/x86_64-pc-linux-gnu/g***/Linux/lib -lsz -ljpeg -L/__w/ESMA-Baselibs/x86_64-pc-linux-gnu/g***/Linux/lib -lcurl -lssl -lcrypto -lz -lm -ldl -lm -L/__w/ESMA-Baselibs/x86_64-pc-linux-gnu/g***/Linux/lib -L/__w/ESMA-Baselibs/x86_64-pc-linux-gnu/g***/Linux//lib   -L/__w/ESMA-Baselibs/x86_64-pc-linux-gnu/g***/Linux/lib -o ncpdq ncpdq.o libnco.la -lexpat -lgsl -lm -lnetcdf -lnetcdf -lnetcdf -L/__w/ESMA-Baselibs/x86_64-pc-linux-gnu/g***/Linux/lib /__w/ESMA-Baselibs/x86_64-pc-linux-gnu/g***/Linux/lib/libhdf5_hl.a /__w/ESMA-Baselibs/x86_64-pc-linux-gnu/g***/Linux/lib/libhdf5.a /__w/ESMA-Baselibs/x86_64-pc-linux-gnu/g***/Linux/lib/libz.a /__w/ESMA-Baselibs/x86_64-pc-linux-gnu/g***/Linux/lib/libsz.a /__w/ESMA-Baselibs/x86_64-pc-linux-gnu/g***/Linux/lib/libcurl.a -lmfhdf -...
</code></pre>
<p>Each one of those <code>g***</code> is a <code>gfortran</code> so for some reason “fortran” is changed to “***”…but why? gcc, good; g++, good, gfortran…bad? I mean, is there some “Fortran is not a Bad Word” setting I need to add to my workflow?</p>
<p>Note: things seem to work, but I’m just really baffled by this.</p>

**Comment by otter-computer** [comment, marked as answer]
<p>That is a bit weird. :thinking: Do you have a secret that contains <code>fortran</code> in it? It might be the secret masker that’s doing it.</p>

**Reply by demerphq** [reply]
Could it be that the secrets masker is a bit too over-zealous with the values it masks and is including flag values like "1", "0"? I see the digit '1' being consistently replaced by *** in log files, and this thread is the only explanation I can find. Would that make sense? 

See: https://github.com/Perl/perl5/runs/7904373887?check_suite_focus=true#step:9:580 where you can see 

```
../cpan/Archive-Tar/t/0***_use.t ....................................... ok
```
the file name is ../cpan/Archive-Tar/t/01_use.t and indeed in that log file every place I see *** where i can easily verify what the value should have been the value is '1'.

**Reply by otter-computer** [reply]
If you've set a secret that contains the value `1` then that's very likely, yep!

**Reply by demerphq** [reply]
I see. Thanks. Perhaps single letter values and digits and words like "true" and "false" should be excluded.  :-) /me imagines what would happen if you put the letters 'a' to 'z' in the secrets as values. :-)

**Comment by mathomp4** [comment]
<p><a class="mention" href="/thomasshaped">@thomasshaped</a> Oooh. I did not know that exisited!</p>
<p>Guess what I added to this workflow when I was updating it? Docker Hub Authorization for my image pull because of the whole “limits on anonymous pulls” thing they announced. And want to guess my Docker Hub username? (Not too much of a reveal as it’s my default internet username almost everywhere.)</p>
<p>Though now as I read <a href="https://github.com/actions/virtual-environments/issues/1445" rel="noopener nofollow ugc">https://github.com/actions/virtual-environments/issues/1445</a>, maybe I don’t need to worry? Sounds like GitHub might have reached an agreement that it isn’t needed for public images pulls into GitHub Actions?</p>
<p>Sigh. Thanks. Weird mystery solved. (Though I guess this could make a <em>very</em> inefficient attack on secrets.)</p>

**Comment by otter-computer** [comment]
<p>Oh no way, wild guess but I’m glad it was right! :sweat_smile:</p>
<aside class="quote no-group" data-username="mathomp4" data-post="3" data-topic="148428">
<div class="title">
<div class="quote-controls"></div>
 mathomp4:</div>
<blockquote>
<p>Sounds like GitHub might have reached an agreement that it isn’t needed for public images pulls into GitHub Actions?</p>
</blockquote>
</aside>
<p>I believe this is the case, yep! Public pulls of images should not be affected by these rate limits when used on GitHub-hosted runners.</p>
